bee-tangle/src/vertex.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{MessageRef, VecSet};
+
+use bee_message::{Message, MessageId};
+
+use std::sync::Arc;
+
+#[derive(Clone)]
+pub struct Vertex<T>
+where
+    T: Clone,
+{
+    message: Option<(MessageRef, T)>,
+    children: (VecSet<MessageId>, bool), // Exhaustive flag
```

Comment:
I'd split this for readability

Reply:
I'd rather not. The exhaustiveness flag is always relevant when accessing `children`, so I want to make it as clear as possible that they're mutually-entangled bits of state.

Reply:
So, maybe a new struct?

Reply:
Why? This is accessed from about 2 locations, all within this file. I'd rather not obscure the definition of an already private field for nebulous gains.

Reply:
sure